### PR TITLE
Added support for building jenkins-ci for next version of debian

### PR DESCRIPTION
**Pull request by ieugen**
This is a replacement for https://github.com/jenkinsci/docker-ssh-agent/pull/255 . 

https://wiki.debian.org/DebianBookworm

Debian bookworm is releasing in ~ 10 days. 
Code wise there will be no breaking changes, only bug fixes.

I believe it's time to consider publishing an updated version based on bookworm.
For our use case we need a newer release of buildah. 

### Testing done

make build
make test

Will push and use in our infra soon. 

### Submitter checklist

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by ieugen** [comment]
@dduportal and all: I've created a new PR based on rebased code on another branch so I will have to close this.
Let's continue discussion there.

I do think that we might have to use separate dockerfiles for different Debian releases. 
The reasoning is that build instructions might be different as packages might change ?! 
(just thinking out loud at this point). 

Also, I've been out of the loop and don't know if I should do for the PR to be merged.

File: README.md [hunk 1]
```diff
@@ -79,8 +79,8 @@ The image has several supported configurations, which can be accessed via the fo
 
 `${IMAGE_VERSION}` can be found on the [releases](https://github.com/jenkinsci/docker-ssh-agent/releases) page.
 
-* `latest`, `latest-jdk11`, `jdk11`, `latest-bullseye-jdk11`, `bullseye-jdk11`, `latest-debian-jdk11`, `debian-jdk11`, `${IMAGE_VERSION}`, `${IMAGE_VERSION}-jdk11`, ([Dockerfile](debian/Dockerfile))
-* `latest-jdk17`, `jdk17`, `latest-bullseye-jdk17`, `bullseye-jdk17`,`latest-debian-jdk17`, `debian-jdk17`, `${IMAGE_VERSION}-jdk17`, ([Dockerfile](debian/Dockerfile))
+* `latest`, `latest-jdk11`, `jdk11`, `latest-bullseye-jdk11`, `bullseye-jdk11`, `latest-bookwrom-jdk11`, `bookworm-jdk11`, `latest-debian-jdk11`, `debian-jdk11`, `${IMAGE_VERSION}`, `${IMAGE_VERSION}-jdk11`, ([Dockerfile](debian/Dockerfile))
```
Comment (gounthar) [inline README.md:82]:
```suggestion
* `latest`, `latest-jdk11`, `jdk11`, `latest-bullseye-jdk11`, `bullseye-jdk11`, `latest-bookworm-jdk11`, `bookworm-jdk11`, `latest-debian-jdk11`, `debian-jdk11`, `${IMAGE_VERSION}`, `${IMAGE_VERSION}-jdk11`, ([Dockerfile](debian/Dockerfile))
```

**Comment by gounthar** [comment]
I apologize for not being able to contribute and offer assistance earlier due to my holidays and participating in GSoC.
Now that I have more time available, I would like to help and support you with that PR.
If you still need assistance, please let me know how I can be of help, and we can discuss the areas you'd like me to focus on.

**Comment by dduportal** [comment]
I'm closing this PR as it was superseded by #299 .

Thanks @ieugen for the help here, you did most of the heavy lifting and we are crediting this work with you along with @gounthar !

**Pull request closed without merging**